airflow/executors/base_executor.py

```diff
@@ -144,23 +156,28 @@ def trigger_tasks(self, open_slots):
             key=lambda x: x[1][1],
             reverse=True)
         for _ in range(min((open_slots, len(self.queued_tasks)))):
-            key, (command, _, queue, simple_ti) = sorted_queue.pop(0)
+            key, (command_to_run, _, _, simple_ti) = sorted_queue.pop(0)
             self.queued_tasks.pop(key)
-            self.running[key] = command
+            self.running.add(key)
             self.execute_async(key=key,
-                               command=command,
-                               queue=queue,
+                               command_to_run=command_to_run,
                                executor_config=simple_ti.executor_config)
 
-    def change_state(self, key, state):
+    def change_state(self, key: ExecutorKeyType, state: str):
+        """Changes state of the task."""
         self.log.debug("Changing state: %s", key)
-        self.running.pop(key, None)
+        try:
+            self.running.remove(key)
+        except KeyError:
```

Comment:
hmmm... , do we want to raise an error here?

Reply:
I wanted to keep the original behaviour. pop(key, None) will return None if no key found.